### PR TITLE
Minor fixes for OpenGL and Window

### PR DESCRIPTION
**Pull request by emilwestergren**
* Fixed minor error in shader
* Added some functions for checking OpenGL extensions
* Fixed so the window is not shrinked to screen size when using high resolutions

### CODE REVIEW

File: source/draw/gpu/opengl/Map/OpenGLES3MapPack.ooc [hunk 1]
```diff
@@ -84,19 +86,22 @@ OpenGLES3MapUnpackRgbaToUv: class extends OpenGLES3MapDefault {
 		super(This fragmentSource, context, false, func {
 			this program setUniform("texture0", 0)
 			this program setUniform("sourceSize", this sourceSize)
-			this program setUniform("targetSize", this targetSize)
+			this program setUniform("targetWidth", this targetSize width)
+			startY: Float = (sourceSize height - targetSize height) / sourceSize height as Float
```
Comment (davidhesselbom) [inline source/draw/gpu/opengl/Map/OpenGLES3MapPack.ooc]: startY := ((sourceSize height - targetSize height) as Float) / sourceSize height

File: test/draw/gpu/android/UnpackRgbaTest.ooc [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+//Set UV values
+for(i in 0..yuvSize width * yuvSize height / 2)
+	rgbaPtr[i] = i % 255
+
+window := Window create(yuvSize)
+
+target := window createYuv420Semiplanar(yuvSize) as GpuYuv420Semiplanar
+gpuRgba := window createGpuImage(rasterRgba)
+_unpackRgbaToMonochrome := OpenGLES3MapUnpackRgbaToMonochrome new(window)
```
Comment (davidhesselbom) [inline test/draw/gpu/android/UnpackRgbaTest.ooc]: It makes no sense to underscore local variables.

File: test/draw/gpu/android/UnpackRgbaTest.ooc [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	if(value != i % 255) {
+		Debug print("ERROR: %u expected: %u" format(value, expected))
+		break
+	}
+
+}
+//yuvRaster save("test.png")
+
+
+for(i in 0..100) {
```
Comment (davidhesselbom) [inline test/draw/gpu/android/UnpackRgbaTest.ooc]: `for (`

File: test/draw/gpu/android/UnpackRgbaTest.ooc [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	expected: UInt = 2* i % 255
+	if(value != expected)
+		Debug print("ERROR: %u expected: %u" format(value, expected))
+}
+
+uvPtr := yuvRaster uv buffer pointer
+for(i in 0..1280 * 360) {
+	value: UInt = uvPtr[i]
+	expected: UInt = i % 255
+	if(value != i % 255) {
```
Comment (davidhesselbom) [inline test/draw/gpu/android/UnpackRgbaTest.ooc]: `if (`

File: test/draw/gpu/android/UnpackRgbaTest.ooc [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+for(i in 0..1280*720) {
+	value: UInt = yPtr[i]
+	expected: UInt = 2* i % 255
+	if(value != expected)
+		Debug print("ERROR: %u expected: %u" format(value, expected))
+}
+
+uvPtr := yuvRaster uv buffer pointer
+for(i in 0..1280 * 360) {
```
Comment (davidhesselbom) [inline test/draw/gpu/android/UnpackRgbaTest.ooc]: Why space around `*` here but not above?

**Comment by emilwestergren** [comment]
Updated.

**Pull request merged**